libbeat/autodiscover/autodiscover.go

```diff
@@ -184,7 +199,7 @@ func (a *Autodiscover) handleStart(event bus.Event) bool {
 			continue
 		}
 
-		a.configs[hash] = &reload.ConfigWithMeta{
+		a.configs[eventID][hash] = &reload.ConfigWithMeta{
```

Comment:
With this refactor the use of a hash here is now confusing. It seems like we're still using `uint64` hashes to be able to work with `RunnerList` for de-duplication. However, I don't see how we'd get duplicates if we're now using a synthetic PK for the config.

Maybe `a.configs` should just be a `map[string][]config` and we can drop the `RunnerList`?

Reply:
This kind of goes toward the [other comment](https://github.com/elastic/beats/pull/8851/files/93858bfac5e1b05171c6da2c317a8b4ddae58a7a#r233102229) I made re: ids. If we have multiple instances of the same provider sharing an ID then we need the de-duplication. If we do not, we don't (we could issue a stop/start on a double start). Maybe let's come to a conclusion on that one first.